source/particle/property/interface.cc

```diff
@@ -220,6 +227,20 @@ namespace aspect
       }
 
       template <int dim>
+      UpdateFlags
+      Manager<dim>::get_needed_update_flags () const
+      {
+        UpdateFlags update = update_default;
+        for (typename std::list<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+             p = property_list.begin(); p!=property_list.end(); ++p)
+          {
+            update |= (*p)->get_needed_update_flags();
+          }
+
+        return update;
```

Comment:
since you say that all flags other than the ones specifically allowed are ignored, I would do something like
```
  return (update & (update_default | update_values | update_gradients));
```
with a comment why you're doing this. This way you're erring on the safe side and implement what you document.

Alternatively, you could also assert that each property only returns these flags, and then you don't have to explicitly mask everything else.